editor/components/writing-flow/index.js

```diff
@@ -98,6 +99,41 @@ class WritingFlow extends Component {
 		} );
 	}
 
+	isInLastNonEmptyBlock( target ) {
+		const tabbables = this.getVisibleTabbables();
+
+		// Find last tabbable, compare with target
+		const lastTabbable = last( tabbables );
+		if ( ! lastTabbable || ! lastTabbable.contains( target ) ) {
+			return false;
+		}
+
+		// Find block-level ancestor of said last tabbable
+		const blockEl = lastTabbable.closest( '.' + BlockListBlock.className );
+		const blockIndex = tabbables.indexOf( blockEl );
+
+		// Unexpected, so we'll leave quietly.
+		if ( blockIndex === -1 ) {
+			return false;
+		}
+
+		// Maybe there are no descendants, and the target is the block itself?
+		if ( lastTabbable === blockEl ) {
+			return isElementNonEmpty( blockEl );
+		}
+
+		// Find descendants of ancestor, i.e. the target and its siblings
+		const blockDescendants = tabbables
+			.slice( blockIndex + 1 )
+			.filter( ( el ) => blockEl.contains( el ) );
+
+		return blockDescendants.some( isElementNonEmpty );
```

Comment:
Minor: Seems an optimization could check both `blockEl.contains` and `isElementNonEmpty` in the same `Array#some`, instead of the separate `Array#filter`.